### PR TITLE
Fix file data not getting picked up correctly, added a few tests

### PR DESCRIPTION
**Pull request by Ricoch**
Tasks:
- Changed base64 data extraction from data received.
- Added some extra tests

### CODE REVIEW

File: lib/active_storage_support/base64_attach.rb [hunk 1]
```diff
@@ -10,9 +10,12 @@ def attachment_from_data(attachment)
     end
 
     def fill_attachment_data(attachment, base64_data)
-      return unless base64_data.try(:is_a?, String) && base64_data =~ /^data:(.*?);(.*?),(.*)$/
+      return unless base64_data.try(:is_a?, String) && base64_data.strip.start_with?('data')
 
-      attachment[:io] = StringIO.new(Base64.decode64(Regexp.last_match(3)))
+      headers, data = base64_data.split(',')
+      headers =~ /^data:(.*?);(.*?)$/
```
Comment (santib) [inline lib/active_storage_support/base64_attach.rb]: be careful, the `;` is also optional `data:[<mediatype>][;base64],<data>` https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs

Comment (Ricoch) [reply lib/active_storage_support/base64_attach.rb]: Changed!

File: active_storage_base64.gemspec [hunk 2]
```diff
@@ -18,6 +18,8 @@ Gem::Specification.new do |s|
 
   # Dependencies
   s.add_dependency 'rails', '~> 5.2.0', '<= 5.2.2'
+  s.add_dependency 'mime-types', '~> 3.0'
```
Comment (santib) [inline active_storage_base64.gemspec]: oh I just see.. are you using these 2 new dependencies?

Comment (Ricoch) [reply active_storage_base64.gemspec]: Oh, yeah, they were for using the MimeTypes approach from Carfrierwave, but we were gonna let ActiveStorage take care of the type inference, so I can remove these two! Good catch

**Pull request merged**